File: frontend/src/types/item.ts
export type ItemKind = "coin" | "banknote";

export type ItemCondition =
  | "UNC"
  | "AU"
  | "XF"
  | "VF"
  | "F"
  | "VG"
  | "G"
  | "P";

export interface Item {
  id: string;
  collectionId: string;
  seriesId: string;
  kind: ItemKind;

  nominal: string;
  currency: string;
  country: string;
  year: number;

  condition: ItemCondition;
  count: number;
  yearFrom: number;
  yearTo: number;

  purchasePrice: number | null;
  purchaseCurrency: string | null;
  purchaseDate: string | null;

  notes: string;

  // «в коллекции» (true) / «в поиске» (false). Заполняется бэком (см. CRUD_CONTRACT.md, раздел 4).
  // Опционален, пока бэк не добавил IsOwned в ItemDto.
  isOwned?: boolean;

  obverseUrl: string | null;
  reverseUrl: string | null;
}

File: EBonistika.API/Entities/Item.cs
namespace EBonistika.API.Entities;

public class Item
{
    public Guid Id { get; set; }
    public Guid CollectionId { get; set; }
    public Guid SeriesId { get; set; }
    public string Kind { get; set; } = string.Empty;
    public string Nominal { get; set; } = string.Empty;
    public string Currency { get; set; } = string.Empty;
    public string Country { get; set; } = string.Empty;
    public int Year { get; set; }
    public string Condition { get; set; } = string.Empty;
    public int Count { get; set; }
    public int YearFrom { get; set; }
    public int YearTo { get; set; }
    public decimal? PurchasePrice { get; set; }
    public string? PurchaseCurrency { get; set; }
    public DateTimeOffset? PurchaseDate { get; set; }
    public string Notes { get; set; } = string.Empty;
    public string? ObverseUrl { get; set; }
    public string? ReverseUrl { get; set; }

    public Collection Collection { get; set; } = null!;
    public Series Series { get; set; } = null!;
}
